### PR TITLE
Use base.tokenize in dask imperative

### PR DESCRIPTION
**Pull request by jcrist**
Allows for more robust deterministic names for imperative.

### CODE REVIEW

**Comment by jcrist** [comment]
Merging soon if no comment.

File: dask/bag/core.py [hunk 1]
```diff
@@ -801,6 +801,10 @@ def to_dataframe(self, columns=None):
                             name, columns, divisions)
 
 
+normalize_token.register(Item, lambda a: a.key)
+normalize_token.register(Bag, lambda a: a.name)
```
Comment (cowlicks) [inline dask/bag/core.py]: How is this related to this PR?

Comment (jcrist) [inline dask/bag/core.py]: Using `base.tokenize` in imperative required that `normalize_token` knew how to handle all collections.
